src/tasks/wifi.c

```diff
@@ -271,22 +274,98 @@ bool wifi_init_task(const int wifi_task_priority,
         return true;
 }
 
+/**
+ * Wrapper for the driver to update the client wifi config
+ * settings.  This is here because it makes sense for these
+ * calls to enter the Wifi subsystem here instead of through
+ * the driver directly.
+ */
+bool wifi_update_client_config(struct wifi_client_cfg *wcc)
+{
+        return esp8266_drv_update_client_cfg(wcc);
+}
+
+/**
+ * Wrapper for the driver to update the ap wifi config
+ * settings.
+ */
+bool wifi_update_ap_config(struct wifi_ap_cfg *wac)
+{
+        return esp8266_drv_update_ap_cfg(wac);
+}
+
 void wifi_reset_config(struct wifi_cfg *cfg)
 {
         /* For now simply zero this out */
         memset(cfg, 0, sizeof(struct wifi_cfg));
 
+        /* Set some sane values for the AP configuration */
+        strncpy(cfg->ap.ssid, "RaceCapture", ARRAY_LEN(cfg->ap.ssid));
+        strncpy(cfg->ap.password, "racecapture", ARRAY_LEN(cfg->ap.password));
+        cfg->ap.channel = 11;
+        cfg->ap.encryption = ESP8266_ENCRYPTION_WPA2_PSK;
+
         /* Inform the Wifi device that settings may have changed */
         wifi_update_client_config(&cfg->client);
+        wifi_update_ap_config(&cfg->ap);
 }
 
 /**
- * Wrapper for the driver to update the client wifi config
- * settings.  This is here because it makes sense for these
- * calls to enter the Wifi subsystem here instead of through
- * the driver directly.
+ * Validates that a given Wifi Ap Configuration is valid
+ * for use.
+ * @return true if it is, false otherwise.
  */
-bool wifi_update_client_config(struct wifi_client_cfg *wcc)
+bool wifi_validate_ap_config(const struct wifi_ap_cfg *wac)
 {
-        return esp8266_drv_update_client_cfg(wcc);
+        return NULL != wac &&
+                wac->channel > 0 &&
+                wac->channel <= WIFI_MAX_CHANNEL &&
+                wac->encryption <= __ESP8266_ENCRYPTION_MAX;
```

Comment:
It looks like an invalid encryption value (-1) might be viewed as a valid ap config based on this statement. Am I seeing that correctly?